domain-mapping/api/class-darkmatter-domains.php

```diff
@@ -121,9 +161,10 @@ private function _basic_check( $fqdn = '' ) {
 	 * @param  boolean $is_https   HTTPS protocol setting.
 	 * @param  boolean $force      Whether the update should be forced.
 	 * @param  boolean $active     Default is active. Set to false if you wish to add a domain but not make it active.
+	 * @param  integer $type       Domain type. Defaults to `1`, which is "main".
 	 * @return DM_Domain|WP_Error             DM_Domain on success. WP_Error on failure.
 	 */
-	public function add( $fqdn = '', $is_primary = false, $is_https = false, $force = true, $active = true ) {
+	public function add( $fqdn = '', $is_primary = false, $is_https = false, $force = true, $active = true, $type = 1 ) {
```

Comment:
1 is a magic value, which you have crated a constant for elsewhere. So you want to never use 1 directly.
As you're already doing an empty check, you could  simply set it to 0 as it's empty is true as well: 
```suggestion
	public function add( $fqdn = '', $is_primary = false, $is_https = false, $force = true, $active = true, $type = 0 ) {
```
or what I've seen later on in the PR:
```suggestion
	public function add( $fqdn = '', $is_primary = false, $is_https = false, $force = true, $active = true, $type = DM_DOMAIN_TYPE_MAIN ) {
```

Reply:
Excellent catch! Fixed.